public/app/features/alerting/unified/state/AlertingQueryRunner.ts

```diff
@@ -51,11 +51,10 @@ export class AlertingQueryRunner {
   }
 
   async run(queries: AlertQuery[], condition: string) {
-    const empty = initialState(queries, LoadingState.Done);
     const queriesToRun = await this.prepareQueries(queries);
 
     if (queriesToRun.length === 0) {
-      return this.subject.next(empty);
```

Comment:
This is probably the most important change here – before this we were always returning new empty query data with different time ranges (because we transform relative time like `last 10 minutes` to absolute time ranges).